From 7a8abf8bfa9c9df9b83fc79146b090ac82ab3d0d Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Tue, 15 Aug 2017 15:24:14 +0800
Subject: [PATCH] dhcp-client: fix invoke dhclient-script failed on Read-only
 file system

In read-only file system, '/etc' is on the readonly partition,
and '/etc/resolv.conf' is symlinked to a separate writable
partition.

In this situation, we create temp files 'resolv.conf.dhclient-new'
in /tmp dir.

Upstream-Status: Pending

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
Signed-off-by: Yi Zhao <yi.zhao@windriver.com>
---
 client/scripts/linux | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/client/scripts/linux b/client/scripts/linux
index cded38f..516c986 100755
--- a/client/scripts/linux
+++ b/client/scripts/linux
@@ -40,7 +40,7 @@ make_resolv_conf() {
     # DHCPv4
     if [ -n "$new_domain_search" ] || [ -n "$new_domain_name" ] ||
        [ -n "$new_domain_name_servers" ]; then
-        new_resolv_conf=/etc/resolv.conf.dhclient-new
+        new_resolv_conf=/tmp/resolv.conf.dhclient-new
         rm -f $new_resolv_conf
 
         if [ -n "$new_domain_name" ]; then
-- 
2.17.1

